graveler/committed/merge_iterator_test.go

```diff
@@ -265,6 +260,231 @@ func TestMergeSeek(t *testing.T) {
+				gotDiffTypes = append(gotDiffTypes, it.Value().Type)
+
+			}
+			if it.Err() != nil {
+				t.Fatalf("got unexpected error: %v", it.Err())
+			}
+			if diff := deep.Equal(tst.expectedKeys, gotKeys); diff != nil {
+				t.Fatalf("got unexpected keys from merge iterator. diff=%s", diff)
+			}
+			if diff := deep.Equal(tst.expectedIdentities, gotValues); diff != nil {
+				t.Fatalf("got unexpected values from merge iterator. diff=%s", diff)
+			}
+			if diff := deep.Equal(tst.expectedDiffTypes, gotDiffTypes); diff != nil {
+				t.Fatalf("got unexpected diff types from merge iterator. diff=%s", diff)
+			}
+		})
+	}
+}
+
+func TestCompareSeek(t *testing.T) {
+	diffs := []graveler.Diff{
+		testMergeNewDiff(added, "k1", "i1", ""),
+		testMergeNewDiff(removed, "k2", "i2", "i2"),
+		testMergeNewDiff(changed, "k3", "i3a", "i3"),
+		testMergeNewDiff(added, "k4", "i4", ""),
+		testMergeNewDiff(removed, "k5", "i5", "i5"),
+		testMergeNewDiff(changed, "k6", "i6", "i6a"),
+		testMergeNewDiff(added, "k7", "i7", ""),
+		testMergeNewDiff(removed, "k8", "i8", "i8"),
+		testMergeNewDiff(changed, "k9", "i9a", "i9"),
+	}
+	diffIt := testutil.NewDiffIter(diffs)
+	baseKeys := []string{"k2", "k3", "k4", "k6"}
+	base := makeBaseIterator(baseKeys)
+	it := committed.NewCompareIterator(diffIt, base)
+	// expected diffs, +k1, -k2, Chng:k3,+k7, Conf:k9,
+	defer it.Close()
+	tests := []struct {
+		seekTo             string
+		expectedKeys       []string
+		expectedIdentities []string
+		expectedDiffTypes  []graveler.DiffType
+	}{
+		{
+			seekTo:             "k1",
+			expectedKeys:       []string{"k1", "k2", "k3", "k7", "k9"},
+			expectedIdentities: []string{"i1", "i2", "i3a", "i7", "i9a"},
+			expectedDiffTypes:  []graveler.DiffType{added, removed, changed, added, conflict},
+		},
+		{
+			seekTo:             "k2",
+			expectedKeys:       []string{"k2", "k3", "k7", "k9"},
+			expectedIdentities: []string{"i2", "i3a", "i7", "i9a"},
+			expectedDiffTypes:  []graveler.DiffType{removed, changed, added, conflict},
+		},
+
+		{
+			seekTo:             "k3",
+			expectedKeys:       []string{"k3", "k7", "k9"},
+			expectedIdentities: []string{"i3a", "i7", "i9a"},
+			expectedDiffTypes:  []graveler.DiffType{changed, added, conflict},
+		},
+
+		{
+			seekTo:             "k4",
+			expectedKeys:       []string{"k7", "k9"},
+			expectedIdentities: []string{"i7", "i9a"},
+			expectedDiffTypes:  []graveler.DiffType{added, conflict},
+		},
+		{
+			seekTo:             "k8",
+			expectedKeys:       []string{"k9"},
+			expectedIdentities: []string{"i9a"},
+			expectedDiffTypes:  []graveler.DiffType{conflict},
+		},
+		{
+			seekTo:             "k9",
+			expectedKeys:       []string{"k9"},
+			expectedIdentities: []string{"i9a"},
+			expectedDiffTypes:  []graveler.DiffType{conflict},
+		},
+		{
+			seekTo:             "k99",
+			expectedKeys:       nil,
+			expectedIdentities: nil,
+		},
+	}
+	for i, tst := range tests {
+		t.Run(fmt.Sprintf("test%d", i), func(t *testing.T) {
```

Comment:
Can we add a name here?  It makes it easier to understand what each case is testing.